core/pkg/artifacts/manifest.go

```diff
@@ -68,6 +79,78 @@ func NewManifestFromProto(proto *service.ArtifactManifest) (Manifest, error) {
 	return manifest, nil
 }
 
+func ManifestContentsFromFile(path string) (map[string]ManifestEntry, error) {
+	manifestFile, err := os.Open(path)
+	if err != nil {
+		return nil, fmt.Errorf("error opening manifest file: %w", err)
+	}
+	defer manifestFile.Close()
+
+	// Whether or not we successfully decode the manifest, we should clean up the file.
+	defer os.Remove(path)
```

Comment:
wait, you first remove the file and then close the file descriptor?
in go defer statements are executed in reverse order last in first out.